### ISSUE TITLE
TypeError when calling getAccessTokenSilently cache-only while no user is logged in

### ISSUE THREAD

**Issue by JasonWeinzierl** [auth0/auth0-angular#457]
### Checklist

- [X] The issue can be reproduced in the [auth0-angular sample app](https://github.com/auth0-samples/auth0-angular-samples/tree/master/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-angular#readme), [Examples](https://github.com/auth0/auth0-angular/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-angular/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-angular/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-angular/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

The following causes a TypeError when no user is logged in:

```
    this.auth.getAccessTokenSilently({ cacheMode: 'cache-only' }).subscribe();
```

I would expect either a more intentional error to be thrown, or change the type annotations to indicate `string | undefined` and return undefined to indicate no access token is available.

### Reproduction

1. Add the above line to HomeComponent' constructor in the auth0-angular sample app
2. Launch the app and don't log in
3. See TypeError logged in the browser console

### Additional context

```
home.component.ts:11 ERROR TypeError: Cannot read properties of undefined (reading 'access_token')
    at TapSubscriber._tapNext (auth0-auth0-angular.mjs:396:137)
    at TapSubscriber._next (tap.js:40:27)
    at TapSubscriber.next (Subscriber.js:49:18)
    at MergeMapSubscriber.notifyNext (mergeMap.js:70:26)
    at SimpleInnerSubscriber._next (innerSubscribe.js:10:21)
    at SimpleInnerSubscriber.next (Subscriber.js:49:18)
    at subscribeToPromise.js:5:24
    at _ZoneDelegate.invoke (zone.js:368:26)
    at Object.onInvoke (core.mjs:26249:33)
    at _ZoneDelegate.invoke (zone.js:367:52)
```

### auth0-angular version

2.2.0

### Angular version

16.1.2

### Which browsers have you tested in?

Chrome, Firefox

**Comment by JasonWeinzierl** [comment]
Probably caused by [typeof token === 'string' returning false when it's undefined, and so token.access_token blows up](https://github.com/auth0/auth0-angular/blob/e711e3c857a9e2a8216f0389d5e473e3f6820294/projects/auth0-angular/src/lib/auth.service.ts#L247C16-L247C16)

**Comment by frederikprijck** [comment]
Thanks for reporting this, I fixed it in #458

**Comment by frederikprijck** [comment]
Fixed and released in [v2.2.1](https://github.com/auth0/auth0-angular/releases/tag/v2.2.1)
